CMakeLists.txt

```diff
@@ -242,14 +242,22 @@ if(GINKGO_BUILD_MPI)
         set(GINKGO_HAVE_GPU_AWARE_MPI OFF)
     endif()
 
-    try_run(uses_openmpi gko_result_unused
+    # use try_compile instead of try_run to prevent cross-compiling issues
+    try_compile(uses_openmpi
         ${Ginkgo_BINARY_DIR}
         ${Ginkgo_SOURCE_DIR}/cmake/openmpi_test.cpp
+                COMPILE_DEFINITIONS -DCHECK_HAS_OPEN_MPI=1
         LINK_LIBRARIES MPI::MPI_CXX
-        RUN_OUTPUT_VARIABLE openmpi_version
+                OUTPUT_VARIABLE output
         )
     if(uses_openmpi)
-        if(openmpi_version VERSION_LESS "4.1")
+        try_compile(valid_openmpi_version
+                    ${Ginkgo_BINARY_DIR}
+                    ${Ginkgo_SOURCE_DIR}/cmake/openmpi_test.cpp
+                    COMPILE_DEFINITIONS -DCHECK_OPEN_MPI_VERSION=1
+                    LINK_LIBRARIES MPI::MPI_CXX
+        )
+        if(NOT valid_openmpi_version)
```

Comment:
By the way, how's the bug in OpenMPI v3.*?

Reply:
Not checked and I don't care. We only support openMPI >= 4. All older releases are officially retired.

Reply:
I think the more interesting question is about openMPI 5.0.0, which was released a few days ago. But I guess I will look into that after the release.

Reply:
I see.

Reply:
does OpenMPI support MPI 3.1 after OpenMPI v4?
Because our documentation claim we support MPI3.1+